### PR TITLE
[SYCL] Make SYCL geometrical marray functions overloads instead of templates 

### PR DESCRIPTION
**Pull request by KornevNikita**
It was decided to change all built-in functions from templates to overloads. This patch changes marray part of "4.17.8. Geometric functions".

Non-marray part: https://github.com/intel/llvm/pull/10014

Spec: https://github.com/KhronosGroup/SYCL-Docs/pull/428

### CODE REVIEW

File: sycl/include/sycl/builtins.hpp [hunk 1]
```diff
@@ -48,6 +48,8 @@ template <class T, int N> marray<T, N> to_marray(vec<T, N> x) {
 namespace __sycl_std = __host_std;
 #endif
 
+#define __SYCL_COMMA ,
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/builtins.hpp:51]: Please add `#undef` for it

**Review by dm-vodopyanov — APPROVED** [review]

**Review by bso-intel — APPROVED** [review]

File: sycl/include/sycl/builtins.hpp (hunk 1 above)
Comment (KornevNikita) [reply sycl/include/sycl/builtins.hpp:51]: [f0d710f](https://github.com/intel/llvm/pull/10048/commits/f0d710f252614766a9a6f4d31a22fbd4f8c444c9)

**Comment by dm-vodopyanov** [comment]
Github is failing to receive the status of some pre-commit checks. Looking into these checks, they were completed successfully, so merging this PR.

**Comment by dm-vodopyanov** [comment]
> > Github is failing to receive the status of some pre-commit checks. Looking into these checks, they were completed successfully, so merging this PR.
> 
> Please, provide the links to the check logs.

https://github.com/intel/llvm/actions/runs/5375236175/jobs/9751482338
https://github.com/intel/llvm/actions/runs/5375236175/jobs/9751482181
and
https://github.com/intel/llvm/actions/runs/5375236175/jobs/9751809445

**Pull request merged**